lib/spack/spack/binary_distribution.py

```diff
@@ -1798,9 +1804,22 @@ def _tar_strip_component(tar: tarfile.TarFile, prefix: str):
             continue
         m.name = m.name[result.end() :]
         if m.linkname:
-            result = regex.match(m.linkname)
-            if result:
-                m.linkname = m.linkname[result.end() :]
+            if m.type == tarfile.SYMTYPE:
+                result = regex.match(m.linkname)
+                is_abs = os.path.isabs(m.linkname)
+                if result or is_abs:
+                    # absolute path outside of the install prefix
+                    # this cannot be extracted because we have no garuntee
+                    # the absolute path exists on the target system
+                    source = m.name
+                    target = m.linkname
+                    tty.warn(
+                        f'Symbolic link from "{source}" to "{target}" cannot be relocated'
+                    )
+            elif m.type == tarfile.LNKTYPE:
+                result = regex.match(m.linkname)
+                if result:
+                    m.linkname = m.linkname[result.end() :]
```

Comment:
Symlinks were treated differently because just dropping the prefix was not correct. If you weren't linking to something in the package install root it would be a broken link always.

```
/a/b/c -> /a//b/d
/a/b/d
---
# relocate package b to prefix /e
/e/b/c -> b/d
/e/b/d
# Error /e/b/b/e files does not exist!
```